Address String.Format issues

rpetrusha opened the pull request:
# Address String.Format issues

This PR corrects a number of issues that primarily concern the C# repl used for String.Format examples:

- It removes some extraneous text in the String.Format documentation.
- It corrects four examples that failed to compile for various reasons (variables not in scope in the repl, no include for namespace, unnecessary attribute included).
- It changes "This example displays the following output:" to "This example displays output like the following:" for output that may vary depending on the state of the DateTimeFormatInfo or NumberFormatInfo object used by the repl.

## Suggested Reviewers

@BillWagner

Commit:
Address String.Format issues

BillWagner commented on xml/System/String.xml:
```diff
@@ -347,7 +350,7 @@
   
 > [!NOTE]
->  The culture-sensitive sorting and casing rules used in string comparison depend on the version of the .NET Framework. In the [!INCLUDE[net_v45](~/includes/net-v45-md.md)] running on the [!INCLUDE[win8](~/includes/win8-md.md)] operating system, sorting, casing, normalization, and Unicode character information conforms to the Unicode 6.0 standard. On other operating systems, it conforms to the U ...
+>  The culture-sensitive sorting and casing rules used in string comparison depend on the version of the .NET Framework. In the .NET Framrwork [!INCLUDE[net_v45](~/includes/net-v45-md.md)] running on the [!INCLUDE[win8](~/includes/win8-md.md)] operating system, sorting, casing, normalization, and Unicode character information conforms to the Unicode 6.0 standard. On other operating systems, it co ...
```
typo: Framrwork.

BillWagner commented on samples/snippets/csharp/VS_Snippets_CLR_System/system.String.Format/cs/starting3.cs:
```diff
@@ -0,0 +1,26 @@
+   public static void Main()
+   {
+       // <Snippet33>
+       int[] years = { 2013, 2014, 2015 };
```
This sample would be a great example for interpolated strings:

```csharp
int[] years = { 2013, 2014, 2015 };
int[] population = { 1025632, 1105967, 1148203 };
var sb = new StringBuilder();
sb.Append($"{"Year",6} {"Population",15}\n\n");
for (int index = 0; index < years.Length; index++)
     sb.Append($"{years[index],6} {population[index],15:N0}\n");       
Console.WriteLine(sb.ToString());
```

I saw a number of other `{0}` and `{1}` code examples. It might be nice to update those as well.

rpetrusha replied:
@BillWagner,, the problem with using string interpolation in these examples is that eliminates the call to the String.Format method, which is the method that is being documented. I've added a note that string interpolation is the recommended alternative, and possibly I should add an item to the Q&A section that compares String.Format with string interpolation, but I think that we should leave the calls to String.Format.

BillWagner replied:
Good points.

BillWagner reviewed (changes requested):
@rpetrusha Overall, this looks great.

I found one minor typo, and I'd suggest updating the string formatting to use interpolated strings. (One example in particular gets much easier to read).

Commit:
Corrected typo, added Q&A item on interpolated strings

rpetrusha commented:
@BillWagner, I've corrected the typo and added an explanation of why we recommend string interpolation to the Q&A section, if you'd like to take another look.

BillWagner commented:
Thanks @rpetrusha I've approved the PR. You can merge when the build finishes.
